searx/version.py

```diff
@@ -1,26 +1,150 @@
-# -*- coding: utf-8 -*-
-'''
-searx is free software: you can redistribute it and/or modify
-it under the terms of the GNU Affero General Public License as published by
-the Free Software Foundation, either version 3 of the License, or
-(at your option) any later version.
-
-searx is distributed in the hope that it will be useful,
-but WITHOUT ANY WARRANTY; without even the implied warranty of
-MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-GNU Affero General Public License for more details.
-
-You should have received a copy of the GNU Affero General Public License
-along with searx. If not, see < http://www.gnu.org/licenses/ >.
-
-(C) 2013- by Adam Tauber, <asciimoo@gmail.com>
-'''
-
-# version of searx
-VERSION_MAJOR = 1
-VERSION_MINOR = 0
-VERSION_BUILD = 0
-
-VERSION_STRING = "{0}.{1}.{2}".format(VERSION_MAJOR,
-                                      VERSION_MINOR,
-                                      VERSION_BUILD)
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# lint: pylint
+# pylint: disable=missing-function-docstring,missing-module-docstring,missing-class-docstring
+
+# some code is from https://github.com/pypa/setuptools_scm/blob/b0b46fd723f6649a7ba4895b005bc89db1529e22/src/setuptools_scm/utils.py  # pylint: disable=line-too-long
+# ( MIT License )
+
+import os
+import shlex
+import subprocess
+
+
+class ShellException(Exception):
+
```

Comment:
Do we need a exception architecture .. why not using the return code of the subprocess directly?

May be it is all better shown by example: I hope I have time this evening .. then I will implement a alternative just to show how I would implement the functionality.

Reply:
Look at the `get_git_url_and_branch` function: it raises an exception if any of the git calls fail.

Then
```python
    try:
        GIT_URL, GIT_BRANCH = get_git_url_and_branch()
    except ShellException as e:
        ...
```
is straightforward